### ISSUE TITLE
Validator client can start with invalid parameters

### ISSUE THREAD

**Issue by yorickdowne** [prysmaticlabs/prysm#9127]
# 🐞 Bug Report

### Description

The validator client will start when given invalid parameters. Example, note the `platypus`:

```
yorick@ethlinux:~/prysm$ ./prysm.sh validator --wallet-dir=/var/lib/prysm/validator --wallet-password-file=/var/lib/prysm/validator/password.txt --accept-terms-of-use platypus
Latest Prysm version is v1.3.11.
Validator is up to date.
Verifying binary integrity.
validator-v1.3.11-linux-amd64: OK
gpg: Signature made Wed 16 Jun 2021 12:30:45 PM EDT
gpg:                using RSA key 0AE0051D647BA3C1A917AF4072E33E4DF1A5036E
gpg: Good signature from "Preston Van Loon <preston@prysmaticlabs.com>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: 0AE0 051D 647B A3C1 A917  AF40 72E3 3E4D F1A5 036E
Verified /home/yorick/prysm/dist/validator-v1.3.11-linux-amd64 has been signed by Prysmatic Labs.
Starting Prysm validator --wallet-dir=/var/lib/prysm/validator --wallet-password-file=/var/lib/prysm/validator/password.txt --accept-terms-of-use platypus
[2021-06-30 10:01:30]  WARN flags: Running on ETH2 Mainnet
[2021-06-30 10:01:30]  INFO node: Opened validator wallet keymanager-kind=direct wallet=/var/lib/prysm/validator/direct
[2021-06-30 10:01:30]  WARN node: Slashing protection file /var/lib/prysm/validator/direct/validator.db is missing.
If you changed your --wallet-dir or --datadir, please copy your previous "validator.db" file into your current --datadir.
Disregard this warning if this is the first time you are running this set of keys.
[2021-06-30 10:01:30]  INFO node: Checking DB databasePath=/var/lib/prysm/validator/direct
Adding optimizations for validator slashing protection 100% [=========]  [0s:0s]
[2021-06-30 10:01:30]  INFO node: Starting validator node version=Prysm/v1.3.11/9fd5575f65ae6f238462ead00ce4f0b16cd0370c. Built at: 2021-06-16 16:21:05+00:00
[2021-06-30 10:01:30]  WARN validator: You are using an insecure gRPC connection. If you are running your beacon node and validator on the same machines, you can ignore this message. If you want to know how to enable secure connections, see: https://docs.prylabs.network/docs/prysm-usage/secure-grpc
```

In practice, a user ran `sudo -u prysmvalidator /opt/prysm/validator/prysm.sh validator --wallet-dir=/var/lib/prysm/validator --wallet-password-file=/var/lib/prysm/validator/password.txt --accept-terms-of-use    validator client-stats --beacon-node-metrics-url=http://localhost:8080/metrics --validator-metrics-url=http://localhost:8081/metrics --clientstats-api-url=https://beaconcha.in/api/v1/stats/APIKEY/server`, which started a second validator process and contributed to them getting slashed.

## 🌍  Your Environment

**Operating System:**

Ubuntu 20.04

**What version of Prysm are you running? (Which release)**

1.3.11

**Comment by ahadda5** [comment]
In terms of running a second validator process, there is this [PR](https://github.com/prysmaticlabs/prysm/pull/9120) coming soon. And is designed to help users not get slashed if they start a second validator (same wallet/keys) without being aware of a running duplicate. 
As for the extra parameters/flag ; you are correct neither the .sh script nor the base code checks for extra params.  However It is harmless though since it will not be used.   Lets see what the _core-team_ says.

**Comment by terencechain** [comment]
addressed in #9129
